app/main.py

```diff
@@ -1,2 +1,61 @@
+from typing import Any, Hashable
+
+
 class Dictionary:
-    pass
+    LOAD_FACTOR = 2 / 3
+
+    def __init__(self) -> None:
+        self.buckets = [[] for _ in range(8)]
+        self.size = 0
+        self.capacity = len(self.buckets)
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        index = self.count_index(key)
+
+        while self.buckets[index]:
+            key_, value_ = self.buckets[index]
+            if key == key_:
+                self.buckets[index] = (key, value)
+                return
+            index = (index + 1) % self.capacity
+
+        self.size += 1
+        self.buckets[index] = (key, value)
+        self.resize()
+
+    def __getitem__(self, key: Hashable) -> Any:
+        index = self.count_index(key)
+
+        while self.buckets[index]:
+            key_, value = self.buckets[index]
+            if key == key_:
+                return value
+            index = (index + 1) % self.capacity
+
+        raise KeyError
+
+    def __len__(self) -> int:
+        return self.size
+
+    def resize(self) -> None:
+        threshold = self.capacity * self.LOAD_FACTOR
+
+        if self.size <= threshold:
+            return
+
+        self.capacity *= 2
+        old_dictionary = self.buckets
+
+        self.buckets = [[] for _ in range(self.capacity)]
+
+        for element in old_dictionary:
+            if element:
+                key, value = element
+                self.__setitem__(key, value)
+                self.size -= 1
```

Comment:
It is not very explicit to decrease self.size for each element. It would be better to assign it to a zero once.